payment_paypal/indico_payment_paypal/controllers.py

```diff
@@ -107,13 +120,28 @@ class RHPaypalSuccess(RHPaypalIPN):
     """Confirmation message after successful payment"""
 
     def _process(self):
+        # Force check on transaction
+        sleeps=0
+        registration = Registration.query.filter_by(uuid=self.token).first()
+        while ((not registration.is_paid) and (sleeps < 30)):
+            current_plugin.logger.debug('Waiting for transaction registration %s',sleeps)
+            time.sleep(1)
```

Comment:
This is not OK. It hogs a uWSGI worker for up to 30s. Sure, the risk of this causing a significant problem is low, but especially no small deployments not negligible.

Check how it was done in [this plugin](https://github.com/indico/indico-plugins-cern/pull/153/files#diff-3bd673336924bee0efbb767ddc620e6119004b1d2b05e965031abe03e1d6bed7) instead.

Reply:
Usually after 2/3 secs, the while passes over, but I understand your idea and approve it. I'll work in such a way.

Reply:
changed all the logic, following the CERN example.